### DISCUSSION TITLE
add_links指定动态库或静态库

### DISCUSSION THREAD

**Discussion by ChrisCatCP** [xmake-io/xmake#3759] [Ideas]
### 你在什么场景下需要该功能？

系统包管理器安装的库一般都是静态库和动态库在一起的
add_links似乎还没法显式指定链接静态库还是动态库

### 描述可能的解决方案

add_links("ssl",{shared = true})
希望能像这样 指定一个或多个库链接方式为动态链接
cxflags为
-Wl,-Bdynamic -lssl

**Comment by waruqi** [comment]
> 系统包管理器安装的库一般都是静态库和动态库在一起的

走 add_requires ，默认 静态库，`add_requires("xxx", {configs = {shared = true}})` 切动态库，不会同时在一起。

> cxflags为
-Wl,-Bdynamic -lssl

这是 linker flags 不是 compiler flags ，不是用 cxflags 。。自己 `add_ldflags` 配置下就行了。动态库走 add_shflags

**Reply by ChrisCatCP** [reply]
add_ldflags倒是可以 但是会影响后续库的链接  如果用了add_links的话
